### PR TITLE
svelte: Preserve current revision in repo search input

### LINKED ISSUE #62713: Svelte: preserve repo revision in search bar
In the Svelte web app prototype, the revision of the repository is not preserved in the search bar. See attached screen recording for details.

To replicate this behavior:

1. search for `repo:github.com/sourcegraph/sourcegraph rev:at.time(1 month ago)` as a convenient way to get right to a particular revision.
2. click on the link to the repo at the revision.
3. See that the revision is in the address bar.
4. Type slash (`/`) (or click on the search bar) and see that the revision is missing from the search bar.

https://github.com/sourcegraph/sourcegraph/assets/129280/9b492afe-54e4-42aa-8faf-8bc279fbf873

### PR DESCRIPTION
**Pull request by fkling**
Closes #62713 

## Test plan

Manually tested with default revision, selected branch and selected revision.

### CODE REVIEW

File: client/web-sveltekit/src/routes/[...repo=reporev]/RepoSearchInput.svelte [hunk 1]
```diff
@@ -34,6 +37,25 @@
         )
     }
 
+    function formatRevision(revision: string, commitID: string): string {
+        // Do not append revison for the default branch
+        if (!revision) {
+            return ''
+        }
+
+        // If the (URL) revision is or starts with the commit ID, use the abbreviated commit ID
```
Comment (camdencheek) [inline client/web-sveltekit/src/routes/[...repo=reporev]/RepoSearchInput.svelte:46]: Oh, that's a smart way to do it.

**Review by camdencheek — APPROVED** [review]
Tried it out locally, LGTM!

File: client/web-sveltekit/src/routes/[...repo=reporev]/RepoSearchInput.svelte (hunk 1 above)
Comment (fkling) [reply client/web-sveltekit/src/routes/[...repo=reporev]/RepoSearchInput.svelte:46]: I in the React app we only check whether the revision is 40 chars, assume its an commit SHA and abbreviate it. This produces incorrect behavior when there is a branch name with exactly 40 chars. We should fix that... at the same time I wish there was a way to abstract all these details from components. Maybe we should compute the human readable revision in the data loader and simply pass that around.

Comment (fkling) [reply client/web-sveltekit/src/routes/[...repo=reporev]/RepoSearchInput.svelte]: I moved the computation to the data loader. We are already doing something similar for repo name. https://github.com/sourcegraph/sourcegraph/pull/62822/commits/7497104c39bd238e3160c2c8b1b7a6f57db5a4ca

**Comment by taiyab** [comment]
@fkling I think this broke showing suggestions right away as soon as you reveal search via `/`.

**Pull request merged**